examples/tutorials/data/veritas.py

```diff
@@ -0,0 +1,451 @@
+ax_excess.plot(
+    info_table["livetime"].to("h"),
+    info_table["excess"],
+    marker="o",
+)
+
+ax_excess.set_title("Excess")
+ax_excess.set_xlabel("Livetime [h]")
+ax_excess.set_ylabel("Excess events")
+
+ax_sqrt_ts.plot(
+    info_table["livetime"].to("h"),
+    info_table["sqrt_ts"],
+    marker="o",
+)
+
+ax_sqrt_ts.set_title("Significance")
+ax_sqrt_ts.set_xlabel("Livetime [h]")
+ax_sqrt_ts.set_ylabel("Significance [sigma]")
+plt.show()
+
+
+######################################################################
+# Make a spectrum
+# -------------------------
+#
+
+######################################################################
+# Now, we’ll calculate the source spectrum. This uses a forward-folding
+# approach that will assume a given spectrum and fit the counts calculated
+# above to that spectrum in each energy bin specified by the
+# ``energy_axis``.
+#
+# For this reason, it’s important that spectral model be set as closely as
+# possible to the expected spectrum - for the Crab nebula, this is a
+# `~gammapy.modeling.models.LogParabolaSpectralModel`.
+#
+
+spectral_model = LogParabolaSpectralModel(
+    amplitude=3.75e-11 * u.Unit("cm-2 s-1 TeV-1"),
+    alpha=2.45,
+    beta=0.15,
+    reference=1 * u.TeV,
+)
+
+model = SkyModel(spectral_model=spectral_model, name="crab")
+
+datasets.models = [model]
+
+fit_joint = Fit()
+result_joint = fit_joint.run(datasets=datasets)
+
+
+######################################################################
+# The best-fit spectral parameters are shown in this table.
+#
+
+print(datasets.models.to_parameters_table())
+
+
+######################################################################
+# We can inspect how well our data fit the model’s predicted counts in
+# each energy bin.
+#
+
+ax_spectrum, ax_residuals = datasets[0].plot_fit()
+ax_spectrum.set_ylim(0.1, 40)
+plt.show()
+
+
+######################################################################
+# We can now calculate flux points to get a spectrum by fitting the
+# ``result_joint`` model’s amplitude in selected energy bands (defined by
+# ``energy_edges``). We set ``selection_optional = "all"`` in
+# `~gammapy.estimators.FluxPointsEstimator`, which will include a calcuation for the upper
+# limits in bins with a significance :math:`< 2\sigma`.
+#
+# In the case of a non-detection or to obtain better upper limits,
+# consider expanding the scan range for the norm parameter in
+# `~gammapy.estimators.FluxPointsEstimator`. See
+# :doc:`../api/estimators` for more details on how to do this.
+#
+
+fpe = FluxPointsEstimator(
+    energy_edges=np.logspace(-0.7, 1.5, 12) * u.TeV,
+    source="crab",
+    selection_optional="all",
+)
+flux_points = fpe.run(datasets=datasets)
+
+
+######################################################################
+# Now, we can plot our flux points along with the best-fit spectral model.
+#
+
+flux_points_dataset = FluxPointsDataset(data=flux_points, models=datasets.models)
```

Comment:
You dont need the `FluxPointsDataset`
Just 
```
ax = flux_points.plot()
spectral_model.plot(ax=ax, energy_bounds=(0.1, 30)*u.TeV)
spectral_model.plot_error(...)
```
should suffice.

In fact, when you are already fitting the DL4 object, there is no reason to fit the flux points where you have upper limits etc...